packages/flutter/lib/src/material/carousel.dart

```diff
@@ -290,6 +294,20 @@ class CarouselView extends StatefulWidget {
   /// Called when one of the [children] is tapped.
   final ValueChanged<int>? onTap;
 
+  /// Whether the default tap interaction for children is disabled.
+  ///
+  /// If true, tap events for all children are disabled by covering them with an [InkWell].
+  /// This prevents direct interaction with child widgets.
+  ///
+  /// If false, tap events are passed through to the child widgets, allowing them
+  /// to handle interactions directly.
+  ///
+  /// Defaults to true.
+  ///
+  /// Note: Setting this to false while also providing an [onTap] callback will
+  /// throw an assertion error, as these options are mutually exclusive.
```

Comment:
Instead of an assertion error, I'd rather we use a GestureDetector in the case that an `onTap` is provided without the ink well.

Reply:
SGTM 👍

While reading through this file, I've noticed several potential ways to tweak it for improved performance. I think it makes sense for this PR to focus on fixing #154701, and then afterward we can go in and improve the structure.